### PR TITLE
Documentation | Update Okta related docs

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?

Updates Okta related documentation to bring it up to date. Specifically:

- Adding documentation for the "Okta Hosted Login Interception" (`login-page-interception.md`)
- Updating the Native Apps integration guide
- Updating the Web Apps integration guide
- Updating the Sessions documentation

### CODE REVIEW

File: docs/okta/web-apps-integration-guide.md [hunk 1]
```diff
@@ -42,13 +42,20 @@ There is a third type of token, which is optional, and we don't recommend for we
...
 1. A redirect callback URI for OAuth authorization callback
    - This is endpoint the reader is redirected back to after the reader authenticates with Okta as part of the OAuth Authorization Code Flow.
    - This endpoint should handle the authorization code and exchange it for an access token and id token.
    - This could be set to a wildcard, but this isn't recommended as it could be used for phishing attacks.
    - e.g. `https://profile.theguardian.com/oauth/authorization-code/callback`
+   - When using the iframe method (only client side, using `response_type=okta_post_message` parameter) for getting/refreshing tokens, this redirect URI must match the base domain of the application, e.g. `https://profile.theguardian.com`
+2. A list of scopes required for the application
+   - This is the permissions that the application will have when using the OAuth tokens.
+   - e.g. `openid profile email`
+   - More information about scopes can be found in the [Tokens](tokens.md) document which we recommend reading.
```
Comment (kelvin-chappell) [inline docs/okta/web-apps-integration-guide.md:56]:
```suggestion
   - More information about scopes can be found in the [Tokens](tokens.md#scopes) document which we recommend reading.
```

File: docs/okta/web-apps-integration-guide.md [hunk 2]
```diff
@@ -42,13 +42,20 @@ There is a third type of token, which is optional, and we don't recommend for we
...
+   - Access tokens should only have the minimum scopes required for your application to access the APIs it needs to access on behalf of the reader, e.g. members-data-api, discussion-api, etc. The Identity team can provide a list of scopes that you'll need for that. A scope looks like the following: `guardian.members-data-api.complete.read.self.secure`
```
Comment (kelvin-chappell) [inline docs/okta/web-apps-integration-guide.md:57]:
```suggestion
   - Access tokens should only have the minimum scopes required for your application to authorise the APIs it needs to access on behalf of the reader, e.g. members-data-api, discussion-api, etc. The Identity team can provide a list of scopes that you'll need for that. A scope looks like the following: `guardian.members-data-api.complete.read.self.secure`
```

**Review by kelvin-chappell — COMMENTED** [review]
Very clear and comprehensive! 👍 
Might be good to use 'sign in' rather than 'log in' throughout for consistency.

File: docs/okta/sessions.md [hunk 3]
```diff
@@ -27,6 +27,8 @@ This cookie is helpful for SSO (Single Sign-On) as it allows us to identify the
 
 We use a session length of 90 days for this, and will be automatically refreshed when a user goes though an OAuth flow, for example the Authorization Code flow.
 
+Since other domains cannot read this cookie, we set a proxy cookie (the existing `GU_U`) cookie, which can be used to identify whether the user _might_ have an Okta session set.
```
[the remark this answers is not in the document]
Comment (coldlink) [reply docs/okta/sessions.md:30]: That's not entirely true either, a user might've accidentally cleared the `GU_U` cookie on `*.theguardian.com` but might still have the `sid`/`idx` cookie on `profile.theguardian.com`.

I'll update the documentation to clarify all the possible states!

File: docs/okta/login-page-interception.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+From the Okta hosted login page we need to pass the following parameters to our own login page:
+
+- **`fromURI`**: This is the URI to redirect to after the user has logged in and has a session set to complete the Authorization Code Flow from the SDK.
+- **`clientId`**: This is the client ID of the application that is calling the SDK and in turn performing the Authorization Code Flow. This parameter can be used to customise the experience our pages.
+
+In the SDK we're also able to pass additional parameters, which we can use to pass the `activation_token` from the user's email link. This can then be used to direct the user to the create password/welcome page instead of the login page, and complete the authorization code flow from there.
+
+The custom javascript for this can be seen in [`scripts/okta/okta-login.ts`](../../scripts/okta/okta-login.ts). And the custom HTML can be seen in [`scripts/okta/okta-login.html`](../../scripts/okta/okta-login.html).
+
+The code is rendered in the Okta hosted sign in page at `/login/login.htm` or even at
```
[the remark this answers is not in the document]
Comment (coldlink) [reply docs/okta/login-page-interception.md:31]: Good point, not sure what happened there 😅

**Review by kelvin-chappell — DISMISSED** [review]
👍

**Review by kelvin-chappell — APPROVED** [review]

**Pull request merged**